Directory.Build.props

```diff
@@ -16,4 +16,10 @@
     <DotNetUseShippingVersions>true</DotNetUseShippingVersions>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(IsTestProject)' != 'true' and '$(IsSourceProjectPackable)' == 'true'">
+    <!-- IsSourceProjectPackable is used in the CI packing leg to enable IsPackable only on source projects.
+         This needs to be evaluated before importing the Arcade SDK and packaging.targets. -->
+    <IsPackable>true</IsPackable>
+  </PropertyGroup>
```

Comment:
I had another idea that might be cleaner than what we have today. If we pass `IsPackable=true` in as a global property but only build the source projects then we wouldn't not need this block above and none of the project file changes in this PR.

Building only the source projects could be achieved by different means, i.e. adding a Traversal build project or a solution filter file.

Reply:
When you say global property, you mean passing it in the yml command, right? Or did you mean a different way?

I looked at the differences between a traversal project and a solution filter. The main differences are that a traversal build project is better suited for a large project tree and need fine-grained control, while a solution filter is a simpler solution.

On one hand, I don't expect the repo to grow much in the future. I think there's a low chance that we will be migrating more packages, as we already have migrated the ones that had real implementations. So it's not imperative to use a traversal build project.

On the other hand, solution filter files seem to be simple enough, but do not allow complex conditioning like the project, if we ever need it.

I am leaning towards the traversal build project. I'll work on that.

Reply:
> When you say global property, you mean passing it in the yml command, right?

Yes